crates/cubecl-cpp/src/shared/element.rs

```diff
@@ -25,21 +25,35 @@ pub enum Elem<D: Dialect> {
     U32,
     U64,
     Bool,
-    Atomic(AtomicKind),
+    Atomic(AtomicKind<D>),
     _Dialect(std::marker::PhantomData<D>),
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Copy, Hash)]
-pub enum AtomicKind {
+pub enum AtomicKind<D: Dialect> {
     I32,
+    I64,
     U32,
+    U64,
+    F16,
+    BF16,
+    F32,
+    F64,
+    _Dialect(std::marker::PhantomData<D>),
```

Comment:
Is there a reason for the dialect to be included here? Maybe add a comment.

Reply:
I added it because I need it to construct the inner element. alternatively, I could directly print the type names, but then you have doubled up string literals and that seemed more error prone. I think the dialect is better overall, so I'll just add a comment.